Adding in various editorial edits from Hugo (#419)
cfrg/draft-irtf-cfrg-opaque: draft-irtf-cfrg-opaque.md

  into `L` bytes of output keying material.
- Nx: The output size of the `Extract()` function in bytes.

> This specification also makes use of a random-key robust Message Authentication Code

Author: I added the clause, ", defined in {{notable-design-differences}}," to help clarify it. It is a bit weird pointing to a section on notable design differences, but that is indeed where it is now defined.

We could pull it out to its own subsection within Security Considerations, i.e. 

```
## Random-Key Robustness

The random-key robustness property that we rely on for a MAC is as follows: given two random keys k1 and k2, it is infeasible to find a message m such that MAC(k1, m) = MAC(k2, m).
```

it would be a short subsection...

> This specification also makes use of a random-key robust Message Authentication Code

Reviewer: Could we just inline the definition here instead?

  into `L` bytes of output keying material.
- Nx: The output size of the `Extract()` function in bytes.

This specification also makes use of a random-key robust Message Authentication Code
(MAC). The random-key robustness property states that, given two random keys k1 and k2,
it is infeasible to find a message m such that MAC(k1, m) = MAC(k2, m).

> The API and parameters for the random-key robust MAC is as follows:

Reviewer: ```suggestion
The API and parameters for the random-key robust MAC are as follows:
```

  into `L` bytes of output keying material.
- Nx: The output size of the `Extract()` function in bytes.

This specification also makes use of a random-key robust Message Authentication Code

> (MAC), defined in {{notable-design-differences}}, with the following API and parameters:

Reviewer: ```suggestion
(MAC). The random-key robustness property states that, given two random keys k1 and k2, it is infeasible to find a message m such that MAC(k1, m) = MAC(k2, m). The API and parameters for the random-key robust MAC is as follows:
```

- server_ake_state: The `ServerAkeState` defined in {{protocol-3dh}}.

> Both of these states are ephemeral, and should be erased after the protocol completes.

Reviewer: ```suggestion
Both of these states are ephemeral and should be erased after the protocol completes.
```

Finally, note that online guessing attacks (against any aPAKE) can be done from
both the client side and the server side. In particular, a malicious server can
attempt to simulate honest responses to learn the client's password.
While this constitutes an exhaustive online attack, hence as expensive as an

> online guessing attack from the client side, it is completely avoided in
> typical cases where the channel between client and server is authenticated,
> e.g., via TLS. In addition to the above, such a channel provides privacy of user
> information, including identity and envelope values.

Reviewer: ```suggestion
online guessing attack from the client side, it can be mitigated in when the channel
between client and server is authenticated, e.g., using server-authenticated TLS.
In such cases, these online attacks are limited to clients and the authenticated server
itself. Moreover, such a channel provides privacy of user information, including identity
and envelope values.
```

I don't think this statement is true. A server that authenticates over TLS can still carry out this attack. This suggestion helps clarify that authentication only changes who can carry out the attack, but does not remove it entirely.
